Formula/sevenzip.rb

```diff
@@ -0,0 +1,36 @@
+class Sevenzip < Formula
+  desc "7-Zip is a file archiver with a high compression ratio"
+  homepage "https://7-zip.org"
+  url "https://7-zip.org/a/7z2106-src.7z"
+  version "21.06"
+  sha256 "675eaa90de3c6a3cd69f567bba4faaea309199ca75a6ad12bac731dcdae717ac"
+  license all_of: ["LGPL-2.1-or-later", "BSD-3-Clause"]
+
+  on_linux do
+    depends_on "gcc@9" => :build
+  end
+
+  def install
+    cd "CPP/7zip/Bundles/Alone2" do
+      on_macos do
+        suffix = Hardware::CPU.arm? ? "arm64" : "x64"
+        system "make", "-f", "../../cmpl_mac_#{suffix}.mak"
+
+        bin.install "b/m_#{suffix}/7zz"
+      end
+
+      on_linux do
+        system "make", "CXX=#{Formula["gcc@9"].opt_bin}/g++-9", "-f", "../../cmpl_gcc.mak"
+
+        bin.install "b/g/7zz"
+      end
```

Comment:
```suggestion
      if OS.mac?
        suffix = Hardware::CPU.intel? ? "x64" : Hardware::CPU.arch
        system "make", "-f", "../../cmpl_mac_#{suffix}.mak"

        bin.install "b/m_#{suffix}/7zz"
      else
        system "make", "-f", "../../cmpl_gcc.mak"

        bin.install "b/g/7zz"
      end
```
We should specify the compiler with the `fails_with` DSL rather than setting `CXX` explicitly.

Reply:
You can probably simplify this with something like
```ruby
mac_suffix = Hardware::CPU.intel? ? "x64" : Hardware::CPU.arch
mk_suffix, directory = if OS.mac?
  ["mac_#{mac_suffix}", "m_#{mac_suffix}"]
else
  ["gcc", "g"]
end

system "make", "-f", "../../cmpl_#{mk_suffix}.mak"
bin.install "b/#{directory}/7zz"
```

Reply:
Fixed. Thanks!